src/automations/components/AutomationActionInput.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <component :is="input.component" v-bind="input.props" class="automation-action-input" />
+</template>
+
+<script lang="ts" setup>
+  import { computed } from 'vue'
+  import AutomationActionPauseDeploymentInput from '@/automations/components/AutomationActionPauseDeploymentInput.vue'
+  import AutomationActionResumeDeploymentInput from '@/automations/components/AutomationActionResumeDeploymentInput.vue'
+  import { AutomationAction } from '@/automations/types'
+  import { withProps } from '@/utilities'
+
+  const props = defineProps<{
+    action: Partial<AutomationAction>,
+  }>()
+
+  const emit = defineEmits<{
+    'update:action': [Partial<AutomationAction>],
+  }>()
+
+  const input = computed(() => {
+    switch (props.action.type) {
+      case 'pause-deployment':
+        return withProps(AutomationActionPauseDeploymentInput, {
+          action: props.action,
+          'onUpdate:action': value => emit('update:action', value),
+        })
+      case 'resume-deployment':
+        return withProps(AutomationActionResumeDeploymentInput, {
+          action: props.action,
+          'onUpdate:action': value => emit('update:action', value),
+        })
+      case undefined:
+        throw new Error('AutomationActionInput.vue action.type is undefined')
```

Comment:
This is a little weird. Because of the type `Partial<AutomationAction>` the types say type can be undefined. Which we don't want, and the implementation in nebula-ui makes sure this cannot happen. But the types allow for it. I tried a few ways to fix this but each thing I tried didn't really work and also created complexity. 

So opting for this for now. Matches existing functionality but also makes sure that we get good type errors when adding new actions.